src/main/java/com/example/springboot/dao/BookDAOService.java

```diff
@@ -0,0 +1,188 @@
+import org.springframework.stereotype.Repository;
+
+import com.example.springboot.entity.Book;
+import com.mysql.jdbc.Statement;
+
+/**
+ * This class implement methods from IBookDAOService interface
+ * @author Kohut Dmytro
+ * @version 1.0
+ */
+@Repository
+@ComponentScan
+public class BookDAOService implements IBookDAOService {
+	
+	@Value("${jdbc.url}") private String URL;
+	@Value("${jdbc.username}") private String USERNAME;  
+	@Value("${jdbc.password}") private String PASSWORD;
+	
+	private static final Logger LOGGER = Logger.getLogger(BookDAOService.class);
+	
+	private static final String QUERY_FIND_ALL = "SELECT id, title, author, reader_id FROM `books`;";
+	private static final String QUERY_SELECT_BY_ID = "SELECT id, title, author, reader_id FROM `books` WHERE id=?;";
+	private static final String QUERY_CREATE = "INSERT INTO `books`(title, author, reader_id) VALUES(?, ?, ?);";
+	private static final String QUERY_UPDATE = "UPDATE `books` SET title=?, author=?, reader_id=? WHERE id=?;";
+	private static final String QUERY_DELETE = "DELETE FROM `books` WHERE id=?;";
+	
+	private static final String ID = "id";
+	private static final String TITLE = "title";
+	private static final String AUTHOR = "author";
+	private static final String READER_ID = "reader_id";
+	
+	/**
+	 * This method create and return a connection
+	 * @return Connection
+	 * @throws SQLException 
+	 */
+	private Connection getConnection() throws SQLException {
+		return DriverManager.getConnection(URL, USERNAME, PASSWORD);
+	}
+	
+	/**
+	 * This method create new instance of Book and return it
+	 * @return Book
+	 * @throws SQLException 
+	 */
+	private Book getBook(ResultSet resultSet) throws SQLException {
+		return new Book(resultSet.getInt(ID), resultSet.getString(TITLE), resultSet.getString(AUTHOR), resultSet.getInt(READER_ID));
+	}
+	
+	/**
+	 * This method return all books from database
+	 * @return List<Book>
+	 */
+	@Override
+	public List<Book> findAll() {
+		List<Book> resultList = new ArrayList<>();
+		
+		try (Connection connection = getConnection()) {
+			PreparedStatement statement = connection.prepareStatement(QUERY_FIND_ALL);
+			ResultSet resultSet = statement.executeQuery();
+			
+			while(resultSet.next()) {
+				resultList.add(getBook(resultSet));
+			}
+			
+			return resultList;
+			
+		} catch(SQLException | RuntimeException e) {
+			LOGGER.error(e.getMessage());
+		}
+		
+		return null;
+	}
+	
+	/**
+	 * This method looking for a book in database by given id
+	 * @param id
+	 * @return Book
+	 */
+	@Override
+	public Book selectById(Integer id) {
+		
+		try (Connection connection = getConnection()) {
+			PreparedStatement statement = connection.prepareStatement(QUERY_SELECT_BY_ID);
+			statement.setInt(1, id);
+			ResultSet resultSet = statement.executeQuery();
+			
+			resultSet.next();
+			Book book = getBook(resultSet);
+			
+			return book;
+			
+		} catch (SQLException | RuntimeException e) {
```

Comment:
yep, it is the best practice to swallow all kind of exceptions and don't share it with other world. `My services works without error, as I AM A PROGRAMMER`